pkg/hiveregistrator/util.go

```diff
@@ -0,0 +1,24 @@
+package hiveregistrator
+
+import (
+	"fmt"
+	"strings"
+)
+
+func ClusterNamespaceName(w WorkloadCluster) string {
+	// create a namespace name for a given cluster that is unique within a hive cluster (or set if multiple)
+	// region should not be necessary, but who knows
+	// TODO: any length constraints ? 63 chars ? 253 characterss?
+	// https://kubernetes.io/docs/concepts/overview/working-with-objects/names/
+
+	return strings.ToLower(fmt.Sprintf("aro-%s-%s-%s", w.Subscription, w.Location, w.Name))
```

Comment:
At the very minimum we need `subscriptionId`, `resourceGroupName`, and `resourceName`.

In the same subscription in the same region we can have clusters with the same name, but in different resource groups.

Reply:
Concatenating subscriptionId, resourceGroupName, and resourceName easily exceeds the 63 character limit of a namespace name. Either we drop one ,truncate the name or use a hash/UUID as a namespace name